Typescript compile has errors (using the tsc version in the starter kit)

Nice work on the starter kit!

I add a npm script to run the typscript compiler in `package.json` like this:

``` json
{
  "scripts": {
    "tsc": "tsc -p src",
  }
}
```

Then when I run `npm run tsc` I get a bunch of compile errors. 
There are some things missing in `tsconfig.json`. I'll try to do a PR to show more what I mean.

When the PR is applied these errors are left and I am not sure how to fix them:

```
$ npm run tsc

15 import Router from "./lib/Router";
node_modules/@types/react-router/index.d.ts(15,20): error TS2307: Cannot find module './lib/Router'.
16 import Link from "./lib/Link";
node_modules/@types/react-router/index.d.ts(16,18): error TS2307: Cannot find module './lib/Link'.
21 import Route from "./lib/Route";
node_modules/@types/react-router/index.d.ts(21,19): error TS2307: Cannot find module './lib/Route'.
26 import hashHistory from "./lib/hashHistory";
node_modules/@types/react-router/index.d.ts(26,25): error TS2307: Cannot find module './lib/hashHistory'.
2 import Calendar from 'react-input-calendar';
src/containers/currency-converter-container/components/currency-valuation-header-calendar.tsx(2,22): error TS2307: Cannot find module 'react-input-calendar'.
46 const actions = Object.assign({}, currencyConverterActions);
src/containers/currency-converter-container/index.tsx(46,24): error TS2339: Property 'assign' does not exist on type 'ObjectConstructor'.
```

--- a/package.json
+++ b/package.json
@@ -29,6 +29,7 @@
     "bvd": "npm run build:vendor && npm run deploy",
     "lint": "tslint ./src/**/*.ts[x]",
     "test": "jspm run src/test-runner | faucet",
+    "tsc": "tsc -p src",
     "_jspm-bundle-vendor-dev": "jspm bundle bundle-config.vendor.dev.js temp/vendor.dev.js -d",
     "_jspm-bundle-vendor": "jspm bundle bundle-config.vendor.js dist/vendor.js -ms",
     "post_jspm-bundle-vendor": "cp jspm.config.js jspm_packages/system.js dist/",

--- a/src/tsconfig.json
+++ b/src/tsconfig.json
@@ -1,10 +1,10 @@
 {
   "compileOnSave": false,
   "compilerOptions": {
-    "skipLibCheck": true,
+    "skipLibCheck": false,
     "declaration": false,
     "jsx": "react",
-    "module": "es6",
+    "module": "commonjs",
     "newLine": "LF",
     "noEmit": true,
     "noEmitOnError": true,
@@ -12,7 +12,7 @@
     "noImplicitThis": false,
     "noUnusedLocals": true,
     "noUnusedParameters": true,
-    "noResolve": true,
+    "noResolve": false,
     "outDir": "../out",
     "preserveConstEnums": true,
     "pretty": true,

--- a/src/tsconfig.json
+++ b/src/tsconfig.json
@@ -4,7 +4,7 @@
     "skipLibCheck": false,
     "declaration": false,
     "jsx": "react",
-    "module": "commonjs",
+    "module": "es6",
     "newLine": "LF",
     "noEmit": true,
     "noEmitOnError": true,
@@ -31,7 +31,7 @@
     "allowJs": false,
     "allowSyntheticDefaultImports": true,
     "noImplicitUseStrict": false,
-    "lib": ["es5", "es6", "es2016.array.include", "dom", "es2016.array.include", "es2015.promise", "es2017.object", "es2015.iterable"],
+    "lib": ["es5", "es6", "dom", "es2016.array.include", "es2015.promise", "es2017.object", "es2015.iterable"],
     "strictNullChecks": false
   },
   "include": [

--- a/src/tsconfig.json
+++ b/src/tsconfig.json
@@ -1,6 +1,7 @@
 {
   "compileOnSave": false,
   "compilerOptions": {
+    "skipLibCheck": true,
     "declaration": false,
     "jsx": "react",
     "module": "es6",
@@ -30,7 +31,7 @@
     "allowJs": false,
     "allowSyntheticDefaultImports": true,
     "noImplicitUseStrict": false,
-    "lib": ["es5", "es6", "es2016.array.include"],
+    "lib": ["es5", "es6", "es2016.array.include", "dom", "es2016.array.include", "es2015.promise", "es2017.object", "es2015.iterable"],
     "strictNullChecks": false
   },
   "include": [

